### ISSUE TITLE
training loss looks very strange

### ISSUE THREAD

**Issue by umichyiwan** [tensorflow/models#670]
Hi, @cbfinn. I'm working on the video prediction model. I just finished training with your code and visualized training loss using TensorBoard. However, I just found training loss looks very strange. Did you also get a similar result?
![training_loss](https://cloud.githubusercontent.com/assets/14853238/20470408/60c0e0ae-af76-11e6-9b1e-fae61d00a430.png)

**Comment by prb12** [comment]
@umichyiwan  Please note that this repository is shared across a large number of models.  

When reporting an issue it is necessary to at least provide a description of which model you are referring to, and preferably enough detail to reproduce the problem (OS version, TF versions, command lines, etc.) so that we can triage the issue to the correct people.

**Comment by cbfinn** [comment]
See my explanation in https://github.com/tensorflow/models/issues/553

**Comment by cbfinn** [comment]
(That curve shape is expected. It is caused by scheduled sampling)

**Comment by umichyiwan** [comment]
Hi @cbfinn, I have seen your answer in 553, it's very clear and definitely helpful.
However, the predicted images are very blur, not like videos in your website.
I write the prediction_test.py file by making some changes on your prediction_train.py.
1. set `FLAGS.pretrained_model` as `/tmp/data/model`
2. add`self.gen_images = gen_images` in Model class before L2 loss and PSNR are calculated.
3. write main function like this:
```
def main(unused_argv):
  print 'Constructing models and inputs.'

  with tf.variable_scope('model', reuse=None):
    val_images, val_actions, val_states = build_tfrecord_input(training=True)
    val_model = Model(val_images, val_actions, val_states, FLAGS.sequence_length)

  print 'Constructing saver.'
  # Make saver.
  saver = tf.train.Saver(tf.get_collection(tf.GraphKeys.VARIABLES), max_to_keep=0)

  # Make training session.
  sess = tf.InteractiveSession()

  print 'restore pretrained model'
  saver.restore(sess, FLAGS.pretrained_model)

  # tf.train.start_queue_runners(sess)
  sess.run(tf.initialize_all_variables())

  tf.logging.info('iteration number, cost')

  # Run through validation set.
  feed_dict = {val_model.lr: 0.0, val_model.prefix: 'val', val_model.iter_num: np.float32(99999)}
  _, val_summary_str, gen_imgs = sess.run([val_model.train_op, val_model.summ_op, val_model.gen_images], feed_dict)
  for i in range(32):
    for batch_imgs in gen_imgs:
      img = batch_imgs[i, :].copy()
      img = cv2.resize(img, (128, 128))
      cv2.imshow('img', img)
      cv2.waitKey(200)
```
Is there any problem with my code?

**Comment by prb12** [comment]
@cbfinn  Please can you check that your gihub account is part of the tensorflow org so that issues can be assigned to you?

**Comment by cbfinn** [comment]
@umichyiwan Can you share example gifs?

**Comment by umichyiwan** [comment]
@cbfinn 
![predicted_img_0](https://cloud.githubusercontent.com/assets/14853238/20680329/98bd2d72-b56c-11e6-9ce7-73eeb693c8fb.jpg)
![predicted_img_1](https://cloud.githubusercontent.com/assets/14853238/20680333/98c2d04c-b56c-11e6-9882-4f2d98fa577e.jpg)
![predicted_img_2](https://cloud.githubusercontent.com/assets/14853238/20680330/98be81f4-b56c-11e6-8772-47264f9fdbfd.jpg)
![predicted_img_3](https://cloud.githubusercontent.com/assets/14853238/20680331/98c07f86-b56c-11e6-9a9e-0cba7b5aedb5.jpg)
![predicted_img_4](https://cloud.githubusercontent.com/assets/14853238/20680332/98c0f40c-b56c-11e6-8810-fd05cb18d2b4.jpg)
![val_loss](https://cloud.githubusercontent.com/assets/14853238/20680618/cf4793e0-b56d-11e6-872b-fcd8bf065717.png)

The validation loss is not exactly same with yours, but the final loss is almost the same. So it should not be the problem. Do you see any problem with my test code above? For example is `feed_dict = {val_model.lr: 0.0, val_model.prefix: 'val', val_model.iter_num: np.float32(99999)}` correct?

**Comment by umichyiwan** [comment]
@cbfinn 
BTW, I also visualized original images by 
adding `self.images = images` in Model class and
change `_, val_summary_str, gen_imgs = sess.run([val_model.train_op, val_model.summ_op, val_model.gen_images], feed_dict)` in main function to  `_, val_summary_str, gen_imgs = sess.run([val_model.train_op, val_model.summ_op, val_model.images], feed_dict)` 
The result looks right
![predicted_img_0](https://cloud.githubusercontent.com/assets/14853238/20732313/2513d236-b65c-11e6-8f88-ea54b026a567.jpg)
![predicted_img_1](https://cloud.githubusercontent.com/assets/14853238/20732314/2514527e-b65c-11e6-83d1-7d0a085d846d.jpg)
![predicted_img_2](https://cloud.githubusercontent.com/assets/14853238/20732311/2512dfca-b65c-11e6-906d-54d0d9d5536d.jpg)
![predicted_img_3](https://cloud.githubusercontent.com/assets/14853238/20732310/251276fc-b65c-11e6-9ca8-0a7cc23578aa.jpg)
![predicted_img_4](https://cloud.githubusercontent.com/assets/14853238/20732312/25130982-b65c-11e6-9428-dd47f03157cd.jpg)
What do you think of this?

**Comment by febert** [comment]
I would recommend comparing the training data and the generated data. The best way to do this is by creating gifs. 

[Here](https://github.com/febert/lsdc/blob/master/python/video_prediction/utils_vpred/create_gif.py#L13) is some code which might help you to create gifs.

**Comment by umichyiwan** [comment]
@febert Thanks for your reply.
I used validation data for test prediction network. And since I got a similar validation loss curve, I assume the training process is correct.(I didn't make any changes in codes)
As shown above, my predicted images are obviously incorrect. But similar code visualizing validation images gave me correct results. This phenomenon makes me cast doubt on the model I used in prediction. Basically I have 2 questions:
1. should I use `feed_dict = {val_model.lr: 0.0, val_model.prefix: 'val', val_model.iter_num: np.float32(99999)}`
2. can you provide your correctly trained model so that I can make sure the model I use is correct.

**Comment by cbfinn** [comment]
I think that the code fix in #1558 fixes this issue.

**Comment by karmel** [comment]
Automatically closing due to lack of recent activity. Since this issue is old at this point, please reopen the issue if it still occurs when tried with the latest version of the code. Thank you.
